wemake_python_styleguide/violations/complexity.py

```diff
@@ -1139,3 +1141,50 @@ class TooManyImportedModuleMembersViolation(ASTViolation):
 
     error_template = 'Found too many imported names from a module: {0}'
     code = 235
+
+
+@final
+class TooComplexFormattedStringViolation(ASTViolation):
+    """
+    Forbids ``f`` strings that are too complex.
+
+    A complex format string is defined as use of any formatted value
+    that is not:
+
+    - the value of a variable
+    - the value of a collection through lookup with a variable, number, or
+      string as the key
+    - the return value of a procedure call without arguments
+
+    Related to :class:`~FormattedStringViolation`.
+
+    Reasoning:
+        Complex ``f`` strings are often difficult to understand,
+        making the code less readible. Generally we don't allow
+        ``f`` strings but this violation exists in case the user
+        decides to ignore the general violation.
+
+    Solution:
+        Use ``.format()`` or assign complex expressions to variables
+        before formatting.
+
+    Example::
+
+        # Wrong:
+        f'{reverse("url-name")}?{"&".join("user="+uid for uid in user_ids)}'
+
+        # Correct:
```

Comment:
Let's swap `Correct` and `Wrong` places (this is how we usually order them).
And leave just a single correct example.